### PR TITLE
Discard the entire panel if is newly added

### PR DESCRIPTION
**Pull request by ivanortegaalba**
https://github.com/grafana/grafana/assets/5699976/f230b225-6b07-43da-94f6-22e59217b34e

**Problem**
When a panel is added to a dashboard, "Discard changes" doesn't apply since the changes are the whole panel. So, when clicking on discarding we expect to remove the widget.

**Solution**
* Rename the button to "Discard panel" to remove the panel entirely when editing the just-added panel
* Enable the "Discard" buttons in the unsaved dashboard too. I don't see the point of not having it in new dashboards

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts [hunk 1]
```diff
@@ -135,11 +138,12 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
       }
       if (getLibraryPanel(panel)) {
         this._handleLibraryPanel(panel, (p) => {
-          this._scene.setState({ editPanel: buildPanelEditScene(p) });
+          this._scene.setState({ editPanel: buildPanelEditScene(p, isNewPanel) });
         });
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts:142]: how can we ever have a new library panel when go into panel edit?

Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts:142]: In the current experience, you can't. You're completely right. I can remove this one.

File: public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts [hunk 2]
```diff
@@ -135,11 +138,12 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
       }
       if (getLibraryPanel(panel)) {
         this._handleLibraryPanel(panel, (p) => {
-          this._scene.setState({ editPanel: buildPanelEditScene(p) });
+          this._scene.setState({ editPanel: buildPanelEditScene(p, isNewPanel) });
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts:141]:
```suggestion
          this._scene.setState({ editPanel: buildPanelEditScene(p) });
```

File: public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
   constructor(private _scene: DashboardScene) {}
 
   getKeys(): string[] {
-    return ['inspect', 'viewPanel', 'editPanel', 'editview', 'autofitpanels'];
+    return ['inspect', 'viewPanel', 'editPanel', 'editview', 'autofitpanels', 'isNewPanel'];
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts:31]: why do we need isNewPanel in url state?

File: public/app/features/dashboard-scene/scene/NavToolbarActions.tsx [hunk 4]
```diff
@@ -172,7 +172,7 @@ export function ToolbarActions({ dashboard }: Props) {
               onClick={() => {
                 const id = dashboard.onCreateNewPanel();
                 DashboardInteractions.toolbarAddButtonClicked({ item: 'add_visualization' });
-                locationService.partial({ editPanel: id });
+                locationService.partial({ editPanel: id, isNewPanel: true });
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/NavToolbarActions.tsx:175]: if we updated onCreateNewPanel so also do `this.setState({ editPanel:  buildPanelEditScene(panel, true)} then I think we could skip this locationService call and the URL sync for isNewPanel (which feels wrong as that parm would not work anyway if doing a full page reload, as the panel would not be found)

File: public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts (hunk 3 above)
Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts:31]: > I used the URL to move the state between scene components. When we create the editor, the object stores in the state if the panel is newly created. To get the "isNew" state, we set that parameter from the CTAs that set the editPanel into the URL.

When adding the visualization, we add the panel calling a function from DashboardScene. Right after, we change the URL to set that key as `editPanel`. We do this from the "Add -> Visualization when editing" and "Add visualization" from the empty state. To get that state in the DashboardScene when getting a new value for  `editPanel` we don't have a way to know if it is editing a new panel unless we set it through the URL when changing. 

I didn't have any other idea how to create PanelEditor with this state if it is not using the URL state. Any better idea?

File: public/app/features/dashboard-scene/scene/NavToolbarActions.tsx (hunk 4 above)
Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/scene/NavToolbarActions.tsx:175]: Ok, I see, let me try that!

Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/scene/NavToolbarActions.tsx]: PR updated! Thanks Torkel for the feedback, now it looks much better

**Review by torkelo — APPROVED** [review]

**Pull request merged**